File: patterns/javascript-patterns/pattern43.js
// Pattern 43: ...

// 12345
// ****
// 123
// **
// 1

const n = 5;

for (let i = n; i >= 1; i--) {
    let row = '';

    for (let j = 1; j <= i; j++) {
        if (i % 2 === 0) {
            row += '*';
        } else {
            row += j;
        }
    }

    console.log(row);
}

File: patterns/python-patterns/pattern43.py
# Pattern 43: ...

# 12345
# ****
# 123
# **
# 1

n = 5

for i in range(n, 0, -1):
    for j in range(1, i + 1):
        if i % 2 == 0:
            print("*", end="")
        else:
            print(j, end="")
    print()
